test/bdd/pkg/messaging/msgservice.go

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * /
+ *
+ */
+
+package messaging
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+)
+
+const (
+	errTimeoutWaitingForMsg = "timeout waiting for incoming message"
+)
+
+// genericInviteMsg is the sample generic invite message
+type genericInviteMsg struct {
+	ID      string   `json:"@id"`
+	Type    string   `json:"@type"`
+	Purpose []string `json:"~purpose"`
+	Message string   `json:"message"`
+	From    string   `json:"from"`
+}
+
+// newMessageService returns new message service instance
+func newMessageService(name, msgType string, purpose []string) *msgService {
+	return &msgService{
+		name:     name,
+		msgType:  msgType,
+		purpose:  purpose,
+		msgQueue: make(chan service.DIDCommMsg),
+	}
+}
+
+// msgService is basic message service implementation
+type msgService struct {
+	name     string
+	purpose  []string
+	msgType  string
+	msgQueue chan service.DIDCommMsg
+}
+
+func (m *msgService) Name() string {
+	return m.name
+}
+
+func (m *msgService) Accept(header *service.Header) bool {
+	var purposeMatched, typeMatched = len(m.purpose) == 0, m.msgType == ""
+
+	if purposeMatched && typeMatched {
+		return false
+	}
+
+	for _, purposeCriteria := range m.purpose {
```

Comment:
this code looks familiar?

Reply:
this function is same as the one we use in rest api, I reused that logic for this test message service